### ISSUE TITLE
Upgrade to 8.2.3 with user_webdavauth in use

### ISSUE THREAD

**Issue by canepan** [owncloud/core#23257]
### Steps to reproduce
1. install OwnCloud < 8.2 and enable user_Webdavauth
2. upgrade to 8.2.3
3. try to enable external user support

### Expected behaviour
it should be possible to enable the app

### Actual behaviour
An error message is shown:
An exception occurred while executing 'CREATE TABLE `oc_users_external` (`backend` VARCHAR(128) DEFAULT '' NOT NULL, `uid` VARCHAR(64) DEFAULT '' NOT NULL, `displayname` VARCHAR(64) DEFAULT NULL, PRIMARY KEY(`uid`, `backend`)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'oc_users_external' already exists

### Server configuration
**Operating system**:
CentOS 6 x86_64

**Web server:**
Apache 2.4

**Database:**
MySQL 5.6

**PHP version:**
5.5
**ownCloud version:** (see ownCloud admin page)
8.2.3 (stable)

**Updated from an older ownCloud or fresh install:**
7.0.3 -> 7.0.13 -> 8.0.11 -> 8.1.6 -> 8.2.3

**Where did you install ownCloud from:**
source tar.bz2

**List of activated apps:**

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
Webdav

### Client configuration
**Browser:**
FireFox

**Operating system:**
MacOSX

**Comment by Phiber2000** [comment]
Have a look at https://github.com/owncloud/core/issues/17939#issuecomment-125753191.
Maybe it correlates with https://github.com/owncloud/core/issues/22993, too. Installation types differ.

@canepan Quick fix: Check database priviledges; drop table 'oc_users_external' if empty.

**Comment by canepan** [comment]
I dropped the table and now I'm able tu enable the app, but webdawauth is not included.
Is there a path to upgrade user_webdavauth to 8.2+?

**Comment by Phiber2000** [comment]
Have a look at #19229:

> The app has been moved to the app user_external

**Comment by Phiber2000** [comment]
For future handling it would be nice to know, if the table was empty before removal?!
Did you check that as recommended?

**Comment by RealRancor** [comment]
> Have a look at #19229:
> 
> The app has been moved to the app user_external

Exactly. This app is not shipped anymore and normally should have been removed when doing a correct upgrade by deleting all files and folders except /data/ and /config/.

**Comment by Phiber2000** [comment]
The question remains:
Is the removal/update of the database table ensured by the updater? Or do we have to implement something handling that in normal upgrade path?
Always fixing something like that manually by dropping tables shouldn't be the right way when using included apps only...

**Comment by canepan** [comment]
I'm sorry for the delay.
I looked at the DB dump of the previous version.
Since the table was not present, I dropped it without looking if it was empty.
As a test, I tried to copy the user_webdavauth.php from version 8.1, so probably the table was created at that point.
I then disabled and removed the user_webdavauth.php file, and enabled the user_external app (after dropping the table), but at that point I was unable to authenticate through WebDav, since there is no GUI to specify the URL.
So I searched for user_external in OC apps (here on github), and I found that I have to insert
`  "user_backends" => array (
    0 => array (
            "class"     => "\OCA\user_external\WebDavAuth",
            "arguments" => array('http://Y.Y.Y.Y/'),
    ),
  ),`
in config.php
In the log I find:
{"reqId":"Vugcp8CoUO8AAFT-V00AAAAG","remoteAddr":"X.X.X.X","app":"OC_USER_WEBDAVAUTH","message":"Invalid Url: \"\" ","level":3,"time":"2016-03-15T14:31:03+00:00"}

Notice that the URL reported is empty.

**Comment by Phiber2000** [comment]
That explains the updating issues. So we can move on:

The right syntax is:
``` php
'user_backends' => array(
    array(
        'class' => '\OCA\User_External\WebDAVAuth',
        'arguments' => array('https://example.com/webdav'),
    ),
),
```
(https://github.com/owncloud/apps/tree/stable8.2/user_external)
At the moment I can't see why your variation shouldn't work, too - but just try that.

**Comment by canepan** [comment]
The reason for not working is a wrong syntax in webdavauth.php.
In constructor, there is:
```php
$this->$webDavAuthUrl =$webDavAuthUrl;
```
but it should be:
```php
$this->webDavAuthUrl =$webDavAuthUrl;
```

**Comment by Phiber2000** [comment]
I see your commit owncloud/apps@854e4a5d514fc973d2a9f6e66a07989d47e86bc9.
That should be pushed to stable branches.

**Comment by LukasReschke** [comment]
@karlitschek Backport of https://github.com/owncloud/apps/commit/854e4a5d514fc973d2a9f6e66a07989d47e86bc9 to 9.0?

**Comment by karlitschek** [comment]
ups. please backport :+1:

**Comment by LukasReschke** [comment]
Will be fixed in 8.2.4 and 9.0.1 :smile:
